src/instana/instrumentation/flask/with_blinker.py

```diff
@@ -4,109 +4,138 @@
+        span.set_attribute(ext.HTTP_METHOD, flask.request.method)
+        if "PATH_INFO" in env:
+            span.set_attribute(ext.HTTP_URL, env["PATH_INFO"])
+        if "QUERY_STRING" in env and len(env["QUERY_STRING"]):
+            scrubbed_params = strip_secrets_from_query(
+                env["QUERY_STRING"],
+                agent.options.secrets_matcher,
+                agent.options.secrets_list,
+            )
+            span.set_attribute("http.params", scrubbed_params)
+        if "HTTP_HOST" in env:
+            span.set_attribute("http.host", env["HTTP_HOST"])
+
+        if (
+            hasattr(flask.request.url_rule, "rule")
+            and path_tpl_re.search(flask.request.url_rule.rule) is not None
+        ):
             path_tpl = flask.request.url_rule.rule.replace("<", "{")
             path_tpl = path_tpl.replace(">", "}")
-            span.set_tag("http.path_tpl", path_tpl)
+            span.set_attribute("http.path_tpl", path_tpl)
     except:
-        logger.debug("Flask before_request", exc_info=True)
+        logger.debug("Flask request_started_with_instana", exc_info=True)
 
 
-def request_finished_with_instana(sender, response, **extra):
-    scope = None
+def request_finished_with_instana(
+    sender: flask.app.Flask, response: flask.wrappers.Response, **extra: Any
+) -> None:
+    span = None
     try:
-        if not hasattr(flask.g, 'scope'):
+        if not hasattr(flask.g, "span"):
             return
 
-        scope = flask.g.scope
-        if scope is not None:
-            span = scope.span
+        span = flask.g.span
+        if span is not None:
 
             if 500 <= response.status_code:
                 span.mark_as_errored()
 
-            span.set_tag(ext.HTTP_STATUS_CODE, int(response.status_code))
+            span.set_attribute(ext.HTTP_STATUS_CODE, int(response.status_code))
             extract_custom_headers(span, response.headers, format=False)
 
-            tracer.inject(scope.span.context, opentracing.Format.HTTP_HEADERS, response.headers)
-            response.headers.add('Server-Timing', "intid;desc=%s" % scope.span.context.trace_id)
+            tracer.inject(span.context, Format.HTTP_HEADERS, response.headers)
+            response.headers.add(
+                "Server-Timing", "intid;desc=%s" % span.context.trace_id
+            )
     except:
-        logger.debug("Flask after_request", exc_info=True)
+        logger.debug("Flask request_finished_with_instana", exc_info=True)
     finally:
-        if scope is not None:
-            scope.close()
+        if span and span.is_recording():
+            span.end()
 
 
-def log_exception_with_instana(sender, exception, **extra):
-    if hasattr(flask.g, 'scope') and flask.g.scope is not None:
-        scope = flask.g.scope
-        if scope.span is not None:
-            scope.span.log_exception(exception)
+def log_exception_with_instana(
+    sender: flask.app.Flask, exception: Any, **extra: Any
```

Comment:
`span.record_exception()` expects an `Exception` as an argument.  As this is Python's built-in class, and all exception classes inherent from it, you can use it here.

```suggestion
    sender: flask.app.Flask, exception: Exception, **extra: Any
```

Reply:
Done!